spec/system/support/cuprite_helpers.rb

```diff
@@ -16,7 +16,7 @@ def debug(*args)
   # Use our `Capybara.save_path` to store screenshots with other capybara artifacts
   # (Rails screenshots path is not configurable https://github.com/rails/rails/blob/49baf092439fc74fc3377b12e3334c3dd9d0752f/actionpack/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb#L79)
   def absolute_image_path
-    Rails.root.join("#{Capybara.save_path}/screenshots/#{image_name}.png")
+    Rails.root.join("#{Capybara.save_path}/screenshots/#{image_name.gsub(/\?/, '')}.png")
```

Comment:
Out of interest, I did a quick google to see if there's a Ruby or Rails way of sanitising filenames. Perhaps other characters will come up one day and cause the same problem (for example a slash).

I found ActiveStorage does, but I don't like the idea of depending on that for spec helpers.

Another method is [`String#parameterize`](https://apidock.com/rails/String/parameterize). That looks like a good general solution for this. What do you think?

```suggestion
    Rails.root.join("#{Capybara.save_path}/screenshots/#{image_name.parameterize}.png")
```

Reply:
I agree!